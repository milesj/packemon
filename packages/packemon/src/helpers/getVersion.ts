import fs from 'fs';
import { PackageStructure } from '@boost/common';

export function getVersion(): string {
	return (
		JSON.parse(
			// @ts-expect-error Not typed for URL
			fs.readFileSync(new URL('../../package.json', import.meta.url), 'utf8'),
		) as PackageStructure
	).version;
}
