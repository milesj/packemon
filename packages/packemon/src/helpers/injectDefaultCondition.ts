/* eslint-disable no-param-reassign */

import { PackageExportConditions, PackageExports } from '../types';

export function injectDefaultCondition(exportMap: PackageExports) {
	Object.entries(exportMap).forEach(([path, conditions]) => {
		if (typeof conditions !== 'object') {
			return;
		}

		// Order from most to least down-leveled
		if (!conditions.default) {
			for (const key of [
				'browser',
				'react-native',
				'electron',
				'node',
				'require',
			] as PackageExportConditions[]) {
				if (conditions[key]) {
					conditions.default = conditions[key];
					delete conditions[key];
					break;
				}
			}
		}

		const keys = Object.keys(conditions);

		if (keys.length === 1) {
			exportMap[path] =
				keys[0] === 'default' ? conditions.default : conditions[keys[0] as PackageExportConditions];
		} else {
			exportMap[path] = conditions;
		}
	});
}
