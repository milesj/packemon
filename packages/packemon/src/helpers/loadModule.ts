import { createRequire } from 'module';

const modRequire = createRequire(import.meta.url);

export function loadModule(name: string, message: string): unknown {
	try {
		return modRequire(name);
	} catch {
		throw new Error(`${message} Please install with \`yarn add --dev ${name}\`.`);
	}
}
