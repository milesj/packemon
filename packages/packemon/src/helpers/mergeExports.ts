/* eslint-disable no-param-reassign */

import { PackageExportConditions, PackageExportPaths } from '../types';

export function mergeExports(
	prev: PackageExportPaths | string,
	base: PackageExportPaths | string,
	root: boolean,
): PackageExportPaths | undefined {
	const next = typeof base === 'string' ? { default: base } : base;

	if (typeof prev === 'string') {
		return next;
	}

	Object.entries(next).forEach(([origKey, nextValue]) => {
		const key = origKey as PackageExportConditions;
		const prevValue = prev[key];

		if (!prevValue) {
			prev[key] = nextValue;
		} else if (nextValue) {
			prev[key] = mergeExports(prevValue, nextValue, false);
		}
	});

	// Always include a default export for tooling to work correctly
	const keys = Object.keys(prev);

	if (root && !prev.default) {
		for (const key of [
			'react-native',
			'electron',
			'browser',
			'node',
			'import',
			'require',
		] as PackageExportConditions[]) {
			if (prev[key]) {
				if (keys.length === 1) {
					return prev[key] as PackageExportPaths;
				}

				prev.default = prev[key];
				delete prev[key];

				break;
			}
		}
	}

	return prev;
}
