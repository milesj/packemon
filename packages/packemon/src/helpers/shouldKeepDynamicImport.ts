import { Format, Platform } from '../types';

// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#browser_compatibility
export function shouldKeepDynamicImport(platform: Platform, format: Format): boolean {
	if (format === 'umd') {
		return false;
	}

	switch (platform) {
		case 'node':
			// >= v13.2
			// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import#browser_compatibility
			return true;
		case 'electron':
		case 'browser':
			// >= 2019
			// https://caniuse.com/es6-module-dynamic-import
			return true;
		case 'native':
			// >= RN 0.72
			// https://metrobundler.dev/docs/module-api/#import-dynamic-import
			return true;
		default:
			return false;
	}
}
