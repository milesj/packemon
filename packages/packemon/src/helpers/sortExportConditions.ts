import { PackageExportPaths } from '../types';

// https://nodejs.org/api/packages.html#conditional-exports
const WEIGHTS = {
	types: 10, // Types must always be first
	misc: 20,
	'node-addons': 30,
	node: 31,
	import: 40,
	require: 50,
	default: 100, // Default must be last
};

export function sortExportConditions<T extends PackageExportPaths | string>(paths: T): T {
	if (typeof paths === 'string') {
		return paths;
	}

	const pathsList: { weight: number; key: string; value: PackageExportPaths | string }[] = [];

	Object.entries(paths).forEach(([key, value]) => {
		pathsList.push({
			weight: key in WEIGHTS ? WEIGHTS[key as 'misc'] : WEIGHTS.misc,
			key,
			value: sortExportConditions(value),
		});
	});

	pathsList.sort((a, d) => {
		const diff = a.weight - d.weight;

		return diff === 0 ? a.key.localeCompare(d.key) : diff;
	});

	return Object.fromEntries(pathsList.map((path) => [path.key, path.value])) as T;
}
