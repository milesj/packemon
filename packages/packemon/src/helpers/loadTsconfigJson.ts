import { Path, type PortablePath } from '@boost/common';
import type { FileSystem } from '../FileSystem';
import type { TSConfigStructure } from '../types';
import { loadModule } from './loadModule';

const CACHE = new Map<Path, TSConfigStructure>();

export function loadTsconfigJson(
	path: PortablePath,
	fs: FileSystem,
): TSConfigStructure | undefined {
	const tsconfigJsonPath = Path.create(path);
	const tsconfig = CACHE.get(tsconfigJsonPath);

	if (tsconfig) {
		return tsconfig;
	}

	if (!tsconfigJsonPath.exists()) {
		return undefined;
	}

	const ts = loadModule(
		'typescript',
		'TypeScript is required for config loading.',
	) as typeof import('typescript');

	// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
	const { config, error } = ts.readConfigFile(tsconfigJsonPath.path(), fs.readFile);

	const host = {
		getCanonicalFileName: (fileName: string) => fileName,
		getCurrentDirectory: () => ts.sys.getCurrentDirectory(),
		getNewLine: () => ts.sys.newLine,
	};

	// istanbul ignore next
	if (error) {
		throw new Error(ts.formatDiagnostic(error, host));
	}

	const result = ts.parseJsonConfigFileContent(
		config,
		ts.sys,
		tsconfigJsonPath.parent().path(),
		{},
		tsconfigJsonPath.path(),
	);

	// istanbul ignore next
	if (result.errors.length > 0) {
		throw new Error(ts.formatDiagnostics(result.errors, host));
	}

	CACHE.set(tsconfigJsonPath, result);

	return result;
}
