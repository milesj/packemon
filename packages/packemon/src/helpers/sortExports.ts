import { PackageExports } from '../types';
import { sortExportConditions } from './sortExportConditions';

const WEIGHTS: Record<string, number> = {
	'./package.json': 0, // First
	'.': 100, // Last
};

export function sortExports(exportMap: PackageExports): PackageExports {
	const paths = Object.keys(exportMap);

	paths.sort((a, d) => {
		const diff = (WEIGHTS[a] ?? 10) - (WEIGHTS[d] ?? 10);

		return diff === 0 ? d.length - a.length : diff;
	});

	return Object.fromEntries(paths.map((path) => [path, sortExportConditions(exportMap[path])]));
}
