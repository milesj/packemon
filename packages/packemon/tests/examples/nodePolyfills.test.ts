import { describe } from 'vitest';
import { testExampleOutput } from '../helpers';

describe('Node polyfills', () => {
	testExampleOutput('node-polyfills.ts', 'babel');
	testExampleOutput('node-polyfills.ts', 'swc');
});
