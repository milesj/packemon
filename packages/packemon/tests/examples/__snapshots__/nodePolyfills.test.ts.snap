// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Node polyfills > babel > transforms example test case: node-current-cjs > cjs/index-node-current-cjs.cjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: cjs

'use strict';

const node_events = require('node:events');
const path = require('node:path');
const _interopDefault = e => e && e.__esModule ? e : {
  default: e
};
const path__default = /*#__PURE__*/_interopDefault(path);
const test = path__default.default.join('foo', 'bar');
const emitter = new node_events.EventEmitter();
class Example extends node_events.EventEmitter {
  log() {}
}
exports.Example = Example;
exports.emitter = emitter;
exports.test = test;
//# sourceMappingURL=index-node-current-cjs.cjs.map
"
`;

exports[`Node polyfills > babel > transforms example test case: node-current-cjs > cjs/index-node-current-cjs-wrapper.mjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// This is an MJS wrapper for a sibling CJS file

import data from './index-node-current-cjs.cjs';

export const { Example, emitter, test } = data;"
`;

exports[`Node polyfills > babel > transforms example test case: node-current-lib > lib/index-node-current-lib.js 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: lib

'use strict';

const node_events = require('node:events');
const path = require('node:path');
const _interopDefault = e => e && e.__esModule ? e : {
  default: e
};
const path__default = /*#__PURE__*/_interopDefault(path);
const test = path__default.default.join('foo', 'bar');
const emitter = new node_events.EventEmitter();
class Example extends node_events.EventEmitter {
  log() {}
}
exports.Example = Example;
exports.emitter = emitter;
exports.test = test;
//# sourceMappingURL=index-node-current-lib.js.map
"
`;

exports[`Node polyfills > babel > transforms example test case: node-current-mjs > mjs/index-node-current-mjs.mjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: mjs

import { EventEmitter } from 'node:events';
import path from 'node:path';
const test = path.join('foo', 'bar');
const emitter = new EventEmitter();
class Example extends EventEmitter {
  log() {}
}
export { Example, emitter, test };
//# sourceMappingURL=index-node-current-mjs.mjs.map
"
`;

exports[`Node polyfills > babel > transforms example test case: node-experimental-cjs > cjs/index-node-experimental-cjs.cjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: cjs

'use strict';

const node_events = require('node:events');
const path = require('node:path');
const _interopDefault = e => e && e.__esModule ? e : {
  default: e
};
const path__default = /*#__PURE__*/_interopDefault(path);
const test = path__default.default.join('foo', 'bar');
const emitter = new node_events.EventEmitter();
class Example extends node_events.EventEmitter {
  log() {}
}
exports.Example = Example;
exports.emitter = emitter;
exports.test = test;
//# sourceMappingURL=index-node-experimental-cjs.cjs.map
"
`;

exports[`Node polyfills > babel > transforms example test case: node-experimental-cjs > cjs/index-node-experimental-cjs-wrapper.mjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// This is an MJS wrapper for a sibling CJS file

import data from './index-node-experimental-cjs.cjs';

export const { Example, emitter, test } = data;"
`;

exports[`Node polyfills > babel > transforms example test case: node-experimental-lib > lib/index-node-experimental-lib.js 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: lib

'use strict';

const node_events = require('node:events');
const path = require('node:path');
const _interopDefault = e => e && e.__esModule ? e : {
  default: e
};
const path__default = /*#__PURE__*/_interopDefault(path);
const test = path__default.default.join('foo', 'bar');
const emitter = new node_events.EventEmitter();
class Example extends node_events.EventEmitter {
  log() {}
}
exports.Example = Example;
exports.emitter = emitter;
exports.test = test;
//# sourceMappingURL=index-node-experimental-lib.js.map
"
`;

exports[`Node polyfills > babel > transforms example test case: node-experimental-mjs > mjs/index-node-experimental-mjs.mjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: mjs

import { EventEmitter } from 'node:events';
import path from 'node:path';
const test = path.join('foo', 'bar');
const emitter = new EventEmitter();
class Example extends EventEmitter {
  log() {}
}
export { Example, emitter, test };
//# sourceMappingURL=index-node-experimental-mjs.mjs.map
"
`;

exports[`Node polyfills > babel > transforms example test case: node-legacy-cjs > cjs/index-node-legacy-cjs.cjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: cjs

'use strict';

const events = require('events');
const path = require('path');
const _interopDefault = e => e && e.__esModule ? e : {
  default: e
};
const path__default = /*#__PURE__*/_interopDefault(path);
const test = path__default.default.join('foo', 'bar');
const emitter = new events.EventEmitter();
class Example extends events.EventEmitter {
  log() {}
}
exports.Example = Example;
exports.emitter = emitter;
exports.test = test;
//# sourceMappingURL=index-node-legacy-cjs.cjs.map
"
`;

exports[`Node polyfills > babel > transforms example test case: node-legacy-cjs > cjs/index-node-legacy-cjs-wrapper.mjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// This is an MJS wrapper for a sibling CJS file

import data from './index-node-legacy-cjs.cjs';

export const { Example, emitter, test } = data;"
`;

exports[`Node polyfills > babel > transforms example test case: node-legacy-lib > lib/index-node-legacy-lib.js 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: lib

'use strict';

const events = require('events');
const path = require('path');
const _interopDefault = e => e && e.__esModule ? e : {
  default: e
};
const path__default = /*#__PURE__*/_interopDefault(path);
const test = path__default.default.join('foo', 'bar');
const emitter = new events.EventEmitter();
class Example extends events.EventEmitter {
  log() {}
}
exports.Example = Example;
exports.emitter = emitter;
exports.test = test;
//# sourceMappingURL=index-node-legacy-lib.js.map
"
`;

exports[`Node polyfills > babel > transforms example test case: node-legacy-mjs > mjs/index-node-legacy-mjs.mjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: mjs

import { EventEmitter } from 'events';
import path from 'path';
const test = path.join('foo', 'bar');
const emitter = new EventEmitter();
class Example extends EventEmitter {
  log() {}
}
export { Example, emitter, test };
//# sourceMappingURL=index-node-legacy-mjs.mjs.map
"
`;

exports[`Node polyfills > babel > transforms example test case: node-stable-cjs > cjs/index-node-stable-cjs.cjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: cjs

'use strict';

const node_events = require('node:events');
const path = require('node:path');
const _interopDefault = e => e && e.__esModule ? e : {
  default: e
};
const path__default = /*#__PURE__*/_interopDefault(path);
const test = path__default.default.join('foo', 'bar');
const emitter = new node_events.EventEmitter();
class Example extends node_events.EventEmitter {
  log() {}
}
exports.Example = Example;
exports.emitter = emitter;
exports.test = test;
//# sourceMappingURL=index-node-stable-cjs.cjs.map
"
`;

exports[`Node polyfills > babel > transforms example test case: node-stable-cjs > cjs/index-node-stable-cjs-wrapper.mjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// This is an MJS wrapper for a sibling CJS file

import data from './index-node-stable-cjs.cjs';

export const { Example, emitter, test } = data;"
`;

exports[`Node polyfills > babel > transforms example test case: node-stable-lib > lib/index-node-stable-lib.js 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: lib

'use strict';

const node_events = require('node:events');
const path = require('node:path');
const _interopDefault = e => e && e.__esModule ? e : {
  default: e
};
const path__default = /*#__PURE__*/_interopDefault(path);
const test = path__default.default.join('foo', 'bar');
const emitter = new node_events.EventEmitter();
class Example extends node_events.EventEmitter {
  log() {}
}
exports.Example = Example;
exports.emitter = emitter;
exports.test = test;
//# sourceMappingURL=index-node-stable-lib.js.map
"
`;

exports[`Node polyfills > babel > transforms example test case: node-stable-mjs > mjs/index-node-stable-mjs.mjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: mjs

import { EventEmitter } from 'node:events';
import path from 'node:path';
const test = path.join('foo', 'bar');
const emitter = new EventEmitter();
class Example extends EventEmitter {
  log() {}
}
export { Example, emitter, test };
//# sourceMappingURL=index-node-stable-mjs.mjs.map
"
`;

exports[`Node polyfills > swc > transforms example test case: node-current-cjs > cjs/index-node-current-cjs.cjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: cjs
"use strict";
const node_events = require("node:events");
const path = require("node:path");
const _interopDefault = (e)=>e && e.__esModule ? e : {
        default: e
    };
const path__default = /*#__PURE__*/ _interopDefault(path);
const test = path__default.default.join("foo", "bar");
const emitter = new node_events.EventEmitter();
let Example = class Example extends node_events.EventEmitter {
    log() {}
};
exports.Example = Example;
exports.emitter = emitter;
exports.test = test;
//# sourceMappingURL=index-node-current-cjs.cjs.map
"
`;

exports[`Node polyfills > swc > transforms example test case: node-current-cjs > cjs/index-node-current-cjs-wrapper.mjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// This is an MJS wrapper for a sibling CJS file

import data from './index-node-current-cjs.cjs';

export const { Example, emitter, test } = data;"
`;

exports[`Node polyfills > swc > transforms example test case: node-current-lib > lib/index-node-current-lib.js 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: lib
"use strict";
const node_events = require("node:events");
const path = require("node:path");
const _interopDefault = (e)=>e && e.__esModule ? e : {
        default: e
    };
const path__default = /*#__PURE__*/ _interopDefault(path);
const test = path__default.default.join("foo", "bar");
const emitter = new node_events.EventEmitter();
let Example = class Example extends node_events.EventEmitter {
    log() {}
};
exports.Example = Example;
exports.emitter = emitter;
exports.test = test;
//# sourceMappingURL=index-node-current-lib.js.map
"
`;

exports[`Node polyfills > swc > transforms example test case: node-current-mjs > mjs/index-node-current-mjs.mjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: current, Format: mjs
import { EventEmitter } from "node:events";
import path from "node:path";
const test = path.join("foo", "bar");
const emitter = new EventEmitter();
let Example = class Example extends EventEmitter {
    log() {}
};
export { Example, emitter, test };
//# sourceMappingURL=index-node-current-mjs.mjs.map
"
`;

exports[`Node polyfills > swc > transforms example test case: node-experimental-cjs > cjs/index-node-experimental-cjs.cjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: cjs
"use strict";
const node_events = require("node:events");
const path = require("node:path");
const _interopDefault = (e)=>e && e.__esModule ? e : {
        default: e
    };
const path__default = /*#__PURE__*/ _interopDefault(path);
const test = path__default.default.join("foo", "bar");
const emitter = new node_events.EventEmitter();
let Example = class Example extends node_events.EventEmitter {
    log() {}
};
exports.Example = Example;
exports.emitter = emitter;
exports.test = test;
//# sourceMappingURL=index-node-experimental-cjs.cjs.map
"
`;

exports[`Node polyfills > swc > transforms example test case: node-experimental-cjs > cjs/index-node-experimental-cjs-wrapper.mjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// This is an MJS wrapper for a sibling CJS file

import data from './index-node-experimental-cjs.cjs';

export const { Example, emitter, test } = data;"
`;

exports[`Node polyfills > swc > transforms example test case: node-experimental-lib > lib/index-node-experimental-lib.js 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: lib
"use strict";
const node_events = require("node:events");
const path = require("node:path");
const _interopDefault = (e)=>e && e.__esModule ? e : {
        default: e
    };
const path__default = /*#__PURE__*/ _interopDefault(path);
const test = path__default.default.join("foo", "bar");
const emitter = new node_events.EventEmitter();
let Example = class Example extends node_events.EventEmitter {
    log() {}
};
exports.Example = Example;
exports.emitter = emitter;
exports.test = test;
//# sourceMappingURL=index-node-experimental-lib.js.map
"
`;

exports[`Node polyfills > swc > transforms example test case: node-experimental-mjs > mjs/index-node-experimental-mjs.mjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: experimental, Format: mjs
import { EventEmitter } from "node:events";
import path from "node:path";
const test = path.join("foo", "bar");
const emitter = new EventEmitter();
let Example = class Example extends EventEmitter {
    log() {}
};
export { Example, emitter, test };
//# sourceMappingURL=index-node-experimental-mjs.mjs.map
"
`;

exports[`Node polyfills > swc > transforms example test case: node-legacy-cjs > cjs/index-node-legacy-cjs.cjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: cjs
"use strict";
const events = require("events");
const path = require("path");
const _interopDefault = (e)=>e && e.__esModule ? e : {
        default: e
    };
const path__default = /*#__PURE__*/ _interopDefault(path);
const test = path__default.default.join("foo", "bar");
const emitter = new events.EventEmitter();
let Example = class Example extends events.EventEmitter {
    log() {}
};
exports.Example = Example;
exports.emitter = emitter;
exports.test = test;
//# sourceMappingURL=index-node-legacy-cjs.cjs.map
"
`;

exports[`Node polyfills > swc > transforms example test case: node-legacy-cjs > cjs/index-node-legacy-cjs-wrapper.mjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// This is an MJS wrapper for a sibling CJS file

import data from './index-node-legacy-cjs.cjs';

export const { Example, emitter, test } = data;"
`;

exports[`Node polyfills > swc > transforms example test case: node-legacy-lib > lib/index-node-legacy-lib.js 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: lib
"use strict";
const events = require("events");
const path = require("path");
const _interopDefault = (e)=>e && e.__esModule ? e : {
        default: e
    };
const path__default = /*#__PURE__*/ _interopDefault(path);
const test = path__default.default.join("foo", "bar");
const emitter = new events.EventEmitter();
let Example = class Example extends events.EventEmitter {
    log() {}
};
exports.Example = Example;
exports.emitter = emitter;
exports.test = test;
//# sourceMappingURL=index-node-legacy-lib.js.map
"
`;

exports[`Node polyfills > swc > transforms example test case: node-legacy-mjs > mjs/index-node-legacy-mjs.mjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: legacy, Format: mjs
import { EventEmitter } from "events";
import path from "path";
const test = path.join("foo", "bar");
const emitter = new EventEmitter();
let Example = class Example extends EventEmitter {
    log() {}
};
export { Example, emitter, test };
//# sourceMappingURL=index-node-legacy-mjs.mjs.map
"
`;

exports[`Node polyfills > swc > transforms example test case: node-stable-cjs > cjs/index-node-stable-cjs.cjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: cjs
"use strict";
const node_events = require("node:events");
const path = require("node:path");
const _interopDefault = (e)=>e && e.__esModule ? e : {
        default: e
    };
const path__default = /*#__PURE__*/ _interopDefault(path);
const test = path__default.default.join("foo", "bar");
const emitter = new node_events.EventEmitter();
let Example = class Example extends node_events.EventEmitter {
    log() {}
};
exports.Example = Example;
exports.emitter = emitter;
exports.test = test;
//# sourceMappingURL=index-node-stable-cjs.cjs.map
"
`;

exports[`Node polyfills > swc > transforms example test case: node-stable-cjs > cjs/index-node-stable-cjs-wrapper.mjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// This is an MJS wrapper for a sibling CJS file

import data from './index-node-stable-cjs.cjs';

export const { Example, emitter, test } = data;"
`;

exports[`Node polyfills > swc > transforms example test case: node-stable-lib > lib/index-node-stable-lib.js 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: lib
"use strict";
const node_events = require("node:events");
const path = require("node:path");
const _interopDefault = (e)=>e && e.__esModule ? e : {
        default: e
    };
const path__default = /*#__PURE__*/ _interopDefault(path);
const test = path__default.default.join("foo", "bar");
const emitter = new node_events.EventEmitter();
let Example = class Example extends node_events.EventEmitter {
    log() {}
};
exports.Example = Example;
exports.emitter = emitter;
exports.test = test;
//# sourceMappingURL=index-node-stable-lib.js.map
"
`;

exports[`Node polyfills > swc > transforms example test case: node-stable-mjs > mjs/index-node-stable-mjs.mjs 1`] = `
"// Bundled with Packemon: https://packemon.dev
// Platform: node, Support: stable, Format: mjs
import { EventEmitter } from "node:events";
import path from "node:path";
const test = path.join("foo", "bar");
const emitter = new EventEmitter();
let Example = class Example extends EventEmitter {
    log() {}
};
export { Example, emitter, test };
//# sourceMappingURL=index-node-stable-mjs.mjs.map
"
`;
